src/main/java/teammates/common/datatransfer/attributes/StudentProfileAttributes.java

```diff
@@ -214,8 +214,8 @@ public Builder withNationality(String nationality) {
             return this;
         }
 
-        public Builder withGender(String gender) {
-            profileAttributes.gender = isGenderValid(gender) ? gender : "other";
+        public Builder withGender(GenderType gender) {
+            profileAttributes.gender = isGenderValid(gender) ? gender : GenderType.other;
```

Comment:
Since you will remove GenderType.INVALID from enums you don't need isGenderValid check anymore. So replace this to a simple assignment and remove isGenderValid method.